### DISCUSSION TITLE
How can I specify temperature for model generation?

### DISCUSSION THREAD

**Discussion by aalyousfi** [outlines-dev/outlines#406] [Q&A] [answered]
Hi,

I'm wondering how I can specify generation parameters like `temperature` for Transformsers or AWQ models? 

I've searched the documentation and the repo but couldn't find anything about that.

**Comment by rlouf** [comment, marked as answer]
Currently not, but we could introduce it after #366. Is that a critical feature for you?

**Reply by aalyousfi** [reply]
@rlouf I think so, I want to set temperature to 0 to make the model more adherent to instructions. I'm wondering what's the default temperature?

After I posted my question, I was going through the repo code. I found [here](https://github.com/outlines-dev/outlines/blob/main/outlines/models/awq.py) that you can pass `model_kwargs` to AWQ models. So if I pass something like `temperature=0.2, do_sample=True`, does that work? Or it has no effect currently too?

**Reply by rlouf** [reply]
This has to be passed to the sampler, so I doubt that `model_kwargs` would change this. This change is planned and tracked by https://github.com/outlines-dev/outlines/issues/335
